### PR TITLE
Workflow Usability Updates

### PR DESCRIPTION
**Pull request by wpoehlm**
This PR adds additional functionality and improves usability to the workflow, including the following:

- The indexed reference genome is now fetched from a synapse folder, rather than being generated by the main workflow (much faster for production runs, but note that this can be slow if you are testing outside of AWS)
- The single end workflow has been updated to support the MSSM dataset (added step to concatenate mapped and unmapped input reads)
- Workflow output files are automatically uploaded to synapse with provenance (An example of what this looks like can be found in syn20783166
- The Toil run script captures the correct github links to use for output provenance
- A utils folder has been added for useful python functions such as generating spot bids on AWS
- A new 'jobs' folder has been added.  This is where you add your input job.json and specify compute resource requirements

A workflow cleanup and improved documentation is in progress

### CODE REVIEW

**Comment by tthyer** [comment]
Note that the tests in this branch were broken when the tools under test were changed to symlinks. I am trying to resolve that issue in a separate branch.

**Comment by jaeddy** [comment]
> Note that the tests in this branch were broken when the tools under test were changed to symlinks. I am trying to resolve that issue in a separate branch.

@tthyer how long do the tests take to run? Would it be worth getting the repo set up with Travis (or GitHub Actions, or whatever)?

**Comment by tthyer** [comment]
@jaeddy I am writing unit tests, so quite fast. Good suggestion on travis -- we already have a JIRA issue to set it up.

**Review by tthyer — APPROVED** [review]
@wpoehlm I pushed two commits to wpoehlmdev. The first to remove the broken testing things that I'm fixing in another branch. The second was for a a few README edits, particularly since the recent provenance changes are not working for me with local runs of cwl-runner or toil, so I made a note that it was still under development in that section.

File: amp-rnaseq_reprocessing/amp-rnaseq_reprocess-workflow/run_readme/Run_Rosmap_9_3_19.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Workflow Submission Preparation
+
+1. Modify main.json
+2. Ensure that reference files have been downloaded to the correct path, as defined in main.json: `./Ref/gencode.v31.annotation.gtf` and `./Ref/GRCh38.primary_assembly.genome.fa` .
+3. Modify any .cwl workflow files to ensure that the resource requirements are appropriate
+4. Modify run-rosmap.py
+5. commit and push changes to git before submitting
```
[the remark this answers is not in the document]
Comment (wpoehlm) [reply amp-rnaseq_reprocessing/amp-rnaseq_reprocess-workflow/run_readme/Run_Rosmap_9_3_19.md]: Sounds good.  When you ran the local cwl-runner test, did you remember to export the environment variables and run with --preserve-entire-environment?  Which reminds me, that needs to be added to the documentation

[the remark this answers is not in the document]
Comment (wpoehlm) [reply amp-rnaseq_reprocessing/amp-rnaseq_reprocess-workflow/run_readme/Run_Rosmap_9_3_19.md]: Never mind the above comment, just saw that you did add that to the readme

**Pull request merged**